packages/compass-aggregations/src/components/pipeline-toolbar/pipeline-settings/pipeline-name.tsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import type { ConnectedProps } from 'react-redux';
+import { css, spacing, Body, Tooltip } from '@mongodb-js/compass-components';
+import type { RootState } from '../../../modules';
+
+const pipelineNameStyles = css({
+  display: 'flex',
+  alignItems: 'center',
+  marginLeft: spacing[1],
+  marginRight: spacing[1],
+});
+
+const nameStyles = css({
+  maxWidth: '100px',
+  whiteSpace: 'nowrap',
+  overflow: 'hidden',
+  textOverflow: 'ellipsis',
+});
+
+const modifiedStyles = css({
+  fontStyle: 'italic',
+});
+
+const PipelineName: React.FunctionComponent<PipelineNameProps> = ({
+  name,
+  isModified,
+}) => {
+  return (
+    <Body className={pipelineNameStyles}>
+      {name === '' ? (
+        'Untitled'
+      ) : (
+        <Tooltip
+          align="top"
+          justify="start"
+          trigger={() => <div className={nameStyles}>{name}</div>}
+        >
+          {name}
+        </Tooltip>
```

Comment:
This tooltip never shows because trigger fn in tooltip component has props that you have to manually pass to the childred when used the way you are using it here